### PR TITLE
Initial feature template module

### PR DESCRIPTION
**Pull request by ganfra**
Initial state of the feature template module.
This includes: 
- gradle file with most used dependencies for a feature module
- main classes : Presenter, View, Events, State and Node
- a unit test class for the Presenter

### CODE REVIEW

File: features/template/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+// TODO: Remove once https://youtrack.jetbrains.com/issue/KTIJ-19369 is fixed
+@Suppress("DSL_SCOPE_VIOLATION")
+plugins {
+    id("io.element.android-compose-library")
+    alias(libs.plugins.ksp)
+    alias(libs.plugins.anvil)
+}
+
+android {
+    namespace = "io.element.android.x.features.template"
```
Comment (bmarty) [inline features/template/build.gradle.kts:26]: Maybe add some TODO everywhere that the developer will have to change something?

File: features/template/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    implementation(project(":anvilannotations"))
+
+    implementation(project(":libraries:di"))
+    implementation(project(":libraries:core"))
+    implementation(project(":libraries:architecture"))
+    implementation(project(":libraries:matrix"))
+    implementation(project(":libraries:matrixui"))
+    implementation(project(":libraries:designsystem"))
+    implementation(project(":libraries:elementresources"))
+    implementation(project(":libraries:ui-strings"))
```
Comment (bmarty) [inline features/template/build.gradle.kts:44]: Maybe create  a `fun DependencyHandlerScope.allLibrariesForFeatures()`? (or reuse `allLibraries()`).

I am not sure if we want to be certain that there is no useless dependencies...

File: features/template/src/main/kotlin/io/element/android/x/features/template/TemplateNode.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import com.bumble.appyx.core.modality.BuildContext
+import com.bumble.appyx.core.node.Node
+import com.bumble.appyx.core.plugin.Plugin
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import io.element.android.x.anvilannotations.ContributesNode
+import io.element.android.x.architecture.presenterConnector
+import io.element.android.x.di.AppScope
+
+// Change to use the right Scope for your feature.
```
Comment (bmarty) [inline features/template/src/main/kotlin/io/element/android/x/features/template/TemplateNode.kt:32]:
```suggestion
// TODO Change to use the right Scope for your feature.
```

**Review by bmarty — APPROVED** [review]
Thanks. Some minor remarks, else LGTM!

File: features/template/build.gradle.kts (hunk 2 above)
Comment (ganfra) [reply features/template/build.gradle.kts:44]: Would be better to avoid using useless dependencies yes. I prefer to let the user pick the right ones

**Pull request merged**